src/output_writers.jl

```diff
@@ -324,3 +324,72 @@ function time_to_write(clock, out::JLD2OutputWriter)
         return false
     end
 end
+
+mutable struct Checkpointer <: OutputWriter
+                 dir :: String
+              prefix :: String
+    output_frequency :: Int
+end
+
+function Checkpointer(; output_frequency, dir=".", prefix="checkpoint", force=false)
+    mkpath(dir)
+    return Checkpointer(dir, prefix, output_frequency)
+end
+
+function savesubfields!(file, model, name, flds=propertynames(getproperty(model, name)))
+    for f in flds
+        file["$name/$f"] = Array(getproperty(getproperty(model, name), f).data.parent)
```

Comment:
Put an `if` statement that does not save a field if its type is `Function`?

Reply:
Sounds like a good check. Won't be needed as long we maintain `checkpointed_fieldsets` but good check either way.

Reply:
I think that we should allow users to supply the `modelfields` that are to be checkpointed. In that case, such a check will be important.

Reply:
Continuing from my comment below, the `if`-statement can also emit a warning that field `x` will not be saved.

Reply:
@ali-ramadhan was this resolved?